api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -157,6 +187,22 @@
     public InputPortMeta getMeta(Operator.InputPort<?> port);
 
     public OutputPortMeta getMeta(Operator.OutputPort<?> port);
+
+    /**
+     * Return collection of stream which are connected to this operator's
+     * input ports.
+     * @param <T>
+     * @return
+     */
+    public <K extends InputPortMeta, V extends StreamMeta> Map<K, V> getInputStreams();
+
+    /**
+     * Return collection of stream which are connected to this operator's
+     * output ports.
+     * @param <T>
+     * @return
+     */
+    public <K extends OutputPortMeta, V extends StreamMeta> Map<K, V> getOutputStreams();
```

Comment:
It is not very obvious when we would need the specific Input and Output streams separately and also not sure the output should in this map format. Can we not include this now and provide this when we come across a use case.